save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/url/ParseUrlInTab.kt

```diff
@@ -0,0 +1,70 @@
+@file:Suppress("FILE_NAME_MATCH_CLASS")
+
+package com.saveourtool.save.frontend.components.views.url
+
+import com.saveourtool.save.domain.Role
+import com.saveourtool.save.entities.benchmarks.TabMenuBar
+import com.saveourtool.save.frontend.components.views.AbstractView
+import com.saveourtool.save.utils.URL_PATH_DELIMITER
+import com.saveourtool.save.validation.FrontendRoutes
+
+import react.State
+
+import kotlinx.browser.window
+
+/**
+ * General interface for working with MenuBar
+ */
+external interface HasSelectedMenu<T> : State {
+    /**
+     * selected value in T Enum
+     */
+    var selectedMenu: T?
+}
+
+/**
+ * The function of analyzing the URL of a tabbed page goes to the tab that was entered in the url, according to the role
+ *
+ * @param menu
+ * @param role
+ * @param flag
+ */
+fun <T, S> AbstractView<*, S>.urlAnalysis(menu: TabMenuBar<T>, role: Role, flag: Boolean?)
```

Comment:
🏴‍☠️?
And why is it nullable? Like a flag for three states?
then use another enum instead of nullable flag
And provide better name for it. I understand that it should indicate something, but I have no idea what it is here for

Reply:
I second this, @Arrgentum please do something about `flag`